kyo-core/shared/src/main/scala/kyo/Duration.scala

```diff
@@ -0,0 +1,74 @@
+package kyo
+
+import java.time.Duration as JDuration
+import java.time.temporal.ChronoUnit.*
+import java.time.temporal.TemporalUnit
+import scala.annotation.switch
+import scala.concurrent.duration.Duration as SDuration
+
+opaque type Duration <: JDuration = JDuration
+given CanEqual[Duration, Duration] = CanEqual.derived
+
+object Duration:
+    val Zero: Duration                   = JDuration.ZERO
+    val Infinity: Duration               = Long.MaxValue.nanos
+    def fromNanos(value: Long): Duration = value.nanos
+    def from(value: JDuration): Duration =
+        (value.nanos: @switch) match
+            case 0                        => Zero
+            case n if n >= Infinity.nanos => Infinity
+            case _                        => value
```

Comment:
Should we reallocate here? Are Java Durations mutable?

Reply:
The javadoc mentions it being immutable and thread safe:

```
 * @implSpec
 * This class is immutable and thread-safe.
 *
 * @since 1.8
 */
@jdk.internal.ValueBased
public final class Duration
```

Reply:
Awesome!